learn/how-tos/implement-openid-okta-provider.md

```diff
@@ -0,0 +1,151 @@
+
+:::note
+Add v1/logout to the end of sign-in redirect URL and enter it as sign-out redirect URL.
+::: 
+
+[![](/learn/assets/wm_openid_ok11.png)](/learn/assets/wm_openid_ok11.png)
+
+12. Select the appropriate access restriction and click **Save**.
+
+[![](/learn/assets/wm_openid_ok12.png)](/learn/assets/wm_openid_ok12.png)
+
+13. Click **Copy** to copy the Client ID to clipboard. Later, provide this information in WaveMaker security configurations. 
+
+[![](/learn/assets/wm_openid_ok13.png)](/learn/assets/wm_openid_ok13.png)
+
+14. Click **Copy** icon to copy the Client Secret to clipboard. Later, provide this information in WaveMaker security configurations. 
+
+[![](/learn/assets/wm_openid_ok14.png)](/learn/assets/wm_openid_ok14.png)
+
+15. Go to **Okta API Scopes** and click **Grant** to include it as scope.
+
+[![](/learn/assets/wm_openid_ok15.png)](/learn/assets/wm_openid_ok15.png)
+
+16. Click **Security** from the menu.
+
+[![](/learn/assets/wm_openid_ok16.png)](/learn/assets/wm_openid_ok16.png)
+
+17. Go to **API** in the drop-down menu.
+
+[![](/learn/assets/wm_openid_ok17.png)](/learn/assets/wm_openid_ok17.png)
+
+18. Click **default**.
+
+[![](/learn/assets/wm_openid_ok18.png)](/learn/assets/wm_openid_ok18.png)
+
+19. Click **Metadata URL** that redirects to the page which contains authorization endpoint, token endpoint, JWKS endpoint and User info endpoint.
+
+[![](/learn/assets/wm_openid_ok19.png)](/learn/assets/wm_openid_ok19.png)
+
+Get the endpoints and client details to provide in the security configuration page in WaveMaker.
+
+
+## Role Mapping using Okta provider:
+
+**Adding Role Attribute**
+1. Go to [Okta Preview](https://developer.okta.com/login/) and set up user account in the Developer console of Okta.
+2. Go to **Applications** > **Applications**.
+3. Check if we have the required web application else we create a new web application.
+
+4. Go to **Directory** > **Profile Editor**.
+5. Click on the **Profile** created by you.
+
+[![](/learn/assets/wm_okta_role2.png)](/learn/assets/wm_okta_role2.png)
+
+6. Click **Add Attribute** and fill in the required attribute details.
+
+[![](/learn/assets/wm_okta_role3.png)](/learn/assets/wm_okta_role3.png)
+
+7. Go to **People** > **Username**.
+
+[![](/learn/assets/wm_okta_role4.png)](/learn/assets/wm_okta_role4.png)
+
+8. Go to **Profile** and click on edit to provide the value for the role attribute.
+
+[![](/learn/assets/wm_okta_role5.png)](/learn/assets/wm_okta_role5.png)
+
+**Creating a custom server as custom claim like role attribute is not allowed in the default server**
+
+9. Go to **Security** > **API** > **Add Authorization Server**.
+
+[![](/learn/assets/wm_okta_role6.png)](/learn/assets/wm_okta_role6.png)
+
+10. Provide the required server details and save the details.
+11. Once the server is created, click on the server.
+
+[![](/learn/assets/wm_okta_role7.png)](/learn/assets/wm_okta_role7.png)
+
+12. Go to **Claims**, add a new claim.
```

Comment:
It would be better to break these points into small sections instead of having them all at one stretch.